lesson13/script.js

```diff
@@ -0,0 +1,83 @@
+function showLoadingImg(){
+    const div = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id = "js-loading";
+    loadingImg.src = "img/loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function fetchdata(){
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    //responseがokではないとき
+    if(!response.ok){
+    throw new Error(`${response.status}:${response.text}`);
+    }
+    //responseがokのとき
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+}
+
+function renderList(result) {
+    const fragment = document.createDocumentFragment();
+    const ul = document.getElementById('js-list');
+        for(const list of result){
+            const li = document.createElement('li');
+            const a = document.createElement('a');
+            const img = document.createElement('img');
+            a.textContent = list.text;
+            a.href = `/${list.a}`;
+            img.src = list.img;
+            img.alt = list.alt;
+            fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+        }
+    ul.appendChild(fragment);
+}
+
+async function init(){
+    showLoadingImg();
+    try{
+        const result = await fetchdata();
+        //resultがtrueではないとき
+        if (!result) {
+            return;
+        }
+        //resultがtrueのとき
+        renderList(result);
+    }catch(e){
+        document.getElementById('js-list').textContent = e;
+        console.error(e);
+    }finally{
+        removeLoadingImg();
+    }
+}
+
+function modalWindow(){
```

Comment:
MUST
I thought the function name should include a verb,
so that people know what it does.

Reply:
Thank you for making me aware of what is important!
I wiil fix.

Reply:
It was difficult to give a clear name to this function name because there are multiple processing contents.
I thought it was too much work to do as a function.
Therefore, we changed to the method of cutting out for each process.
https://github.com/nakagawamai/js-lesson/pull/15/commits/4504f6c0a9ffe7a50f8ff9a41eb1ea58497412dc

Reply:
I commented below :)